src/lsm/lsm_manager.c

```diff
@@ -116,17 +117,18 @@ __lsm_stop_workers(WT_SESSION_IMPL *session)
 {
 	WT_LSM_MANAGER *manager;
 	WT_LSM_WORKER_ARGS *worker_args;
-	uint32_t i;
 
 	manager = &S2C(session)->lsm_manager;
 	/*
-	 * Start at the end of the list of threads and stop them until we
-	 * have the desired number.  We want to keep all active threads
-	 * packed at the front of the worker array.
+	 * Start at the end of the list of threads and stop them until we have
+	 * the desired number. We want to keep all active threads packed at the
+	 * front of the worker array.
 	 */
-	WT_ASSERT(session, manager->lsm_workers != 0);
-	for (i = manager->lsm_workers - 1; i >= manager->lsm_workers_max; i--) {
-		worker_args = &manager->lsm_worker_cookies[i];
+	WT_ASSERT(session, manager->lsm_workers > manager->lsm_workers_max);
```

Comment:
`__lsm_stop_workers()` is only called from `__wt_lsm_manager_reconfig()`, where we've already asserted `manager->lsm_workers_max < the previous manager->lsm_workers_max`.

I only changed the assert because checking `manager->lsm_workers != 0` didn't seem to add anything at all, and the assert I added matched the assert in `__lsm_general_worker_start()`.

I'm happy to remove it if you like.